app/models/subscription_content.rb

```diff
@@ -1,25 +1,12 @@
 class SubscriptionContent < ApplicationRecord
   belongs_to :subscription
-  belongs_to :content_change, optional: true
-  belongs_to :message, optional: true
   belongs_to :digest_run_subscriber, optional: true
   belongs_to :email, optional: true
 
+  # A subscription content should always have one of these and not both
+  belongs_to :content_change, optional: true
+  belongs_to :message, optional: true
+
   scope :immediate, -> { where(digest_run_subscriber_id: nil) }
   scope :digest, -> { where.not(digest_run_subscriber_id: nil) }
-
-  validate :presence_of_content_change_or_message
-
-  def presence_of_content_change_or_message
```

Comment:
Or....[this beauty](https://github.com/alphagov/service-manual-publisher/pull/816/commits/4eff0b9afbcc70ae75b376329725c5e45255e19d) (you'll need to scroll the not-quite-deleted diff at the end to see).